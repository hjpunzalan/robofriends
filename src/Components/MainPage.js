import React, { Component } from 'react';
import CardList from '../Components/CardList.js';
import SearchBox from '../Components/SearchBox';
import Scroll from '../Components/Scroll';
import ErrorBoundary from '../Components/ErrorBoundary';
import Header from '../Components/Header';
import './MainPage.css';

class MainPage extends Component {
	componentDidMount() {
		this.props.onRequestRobots();
	}

	filterRobots = () => {
		return this.props.robots.filter(robot => {
			return robot.name
				.toLowerCase()
				.includes(this.props.searchField.toLowerCase());
		});
	};

	render() {
		const { onSearchChange, isPending } = this.props;
		const wrapper = child => (
			<div className="tc">
				<h1 className="f1"> My Robo Friends</h1>
				<Header />
				<SearchBox searchChange={onSearchChange} />
				<Scroll>{child}</Scroll>
			</div>
		);
		return (
			<wrapper>
				{isPending ? (
					<h1> Loading... </h1>
				) : (
					<ErrorBoundary>
						<CardList robots={this.filterRobots} />
					</ErrorBoundary>
				)}
			</wrapper>
		);
	}
}

export default MainPage;
