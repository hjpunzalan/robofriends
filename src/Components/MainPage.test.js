import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

let wrapper;
// runs before each test as 'middleware'
beforeEach(() => {
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: false
	};
	wrapper = shallow(<MainPage {...mockProps} />);
});

it('renders mainPage without crashing', () => {
	expect(wrapper).toMatchSnapshot();
});

// instance of the wrapper
it('filters robots correctly', () => {
	// props required of main page
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [
			{
				id: 3,
				name: 'John',
				email: 'john@gmail.com'
			}
		],
		searchField: 'john',
		isPending: false
	};
	const wrapper2 = shallow(<MainPage {...mockProps} />);
	expect(wrapper2.instance().filterRobots()).toEqual([
		{
			id: 3,
			name: 'John',
			email: 'john@gmail.com'
		}
	]);
});
// redux-mock-store -- npm package

it('filters robots correctly 2', () => {
	// props required of main page
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [
			{
				id: 3,
				name: 'John',
				email: 'john@gmail.com'
			}
		],
		searchField: 'a',
		isPending: false
	};
	const filteredRobots = [];
	const wrapper3 = shallow(<MainPage {...mockProps} />);
	expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
});

it('is Pending', () => {
	// props required of main page
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [
			{
				id: 3,
				name: 'John',
				email: 'john@gmail.com'
			}
		],
		searchField: 'a',
		isPending: true
	};
	const loading = (
		<wrapper>
			<h1> Loading... </h1>
		</wrapper>
	);
	const wrapper4 = shallow(<MainPage {...mockProps} />);
	expect(wrapper4.instance().render()).toEqual(loading);
});
