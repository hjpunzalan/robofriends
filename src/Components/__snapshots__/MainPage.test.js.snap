// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders mainPage without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    isPending={false}
    onRequestRobots={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    robots={Array []}
    searchField=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ErrorBoundary>
        <CardList
          robots={[Function]}
        />
      </ErrorBoundary>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <CardList
          robots={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "robots": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "wrapper",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ErrorBoundary>
          <CardList
            robots={[Function]}
          />
        </ErrorBoundary>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <CardList
            robots={[Function]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "robots": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "wrapper",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
